lib/util/host-rules.spec.ts

```diff
@@ -235,6 +235,26 @@ describe('util/host-rules', () => {
       expect(find({ url: 'httpsdomain.com' }).token).toBeUndefined();
     });
 
+    it('matches on matchHost with port', () => {
+      add({
+        matchHost: 'https://domain.com:9118',
+        token: 'def',
+      });
+      expect(find({ url: 'https://domain.com:9118' }).token).toBe('def');
+      expect(find({ url: 'https://domain.com' }).token).toBeUndefined();
+      expect(find({ url: 'httpsdomain.com' }).token).toBeUndefined();
+    });
+
+    it('host with port is interpreted as empty', () => {
```

Comment:
Should we instead fix this and massage it to https:// ?

Reply:
Not that easy as in scenarios where users have to define ports it could also happen that they are using HTTP instead of HTTPS. 

I would suggest to merge this, to document the current behavior and then fix this down the line. 
We have to probably use another way to parse the URL.

Reply:
As is, it will work for no one. If we convert to https, it probably works for most. Is there a downside?

Reply:
To decide if we have to massage basically we would have to parse the `matchHost` field with port awareness. 
If we would implement this we could fix it right away rather then massaging.

Reply:
We could just pretend https:// and document that instead